### PR TITLE
Supports passing inputs via the CLI

### PR DESCRIPTION
**Pull request by eahefnawy**
## What has been implemented?
Supports passing inputs via the CLI according to Option C in the following specs:

https://github.com/serverless/roadmap/issues/453
https://github.com/serverless/roadmap-tencent/issues/512

### CODE REVIEW

File: src/cli/utils.js [hunk 1]
```diff
@@ -806,6 +806,61 @@ const validateNodeModules = async (directory) => {
...
+      return value.map((v) => setCliInputValueType(v));
+    }
+
+    // check if the string is actually a number
+    if (!isNaN(value)) {
+      value = Number(value);
+    }
+
+    // check if the string is actually boolean
+    value = value === 'true' || (value === 'false' ? false : value);
```
Comment (timqian) [inline src/cli/utils.js:844]: Will this be an issue when the user actually wants to pass the string "true"?

**Comment by ac360** [comment]
Thx @eahefnawy.  We need object support in here, especially to pass event objects to serverless functions.  Perhaps we can try to auto-convert strings to objects, so users can use `event='{"foo": "bar"}'` like we do in V1?

File: src/cli/utils.js (hunk 1 above)
Comment (eahefnawy) [reply src/cli/utils.js:844]: Yeah, it won't work. What's the use case for that though? For edge cases like this, I think users should just use the YAML file. There's only so much you can do with the CLI.

**Comment by eahefnawy** [comment]
@ac360 hmm ok yeah that's better.

**Comment by timqian** [comment]
I have tested this feature in China, `String`, `Array`, and `JSON` are all parsed correctly as described in the doc.

File: README.md [hunk 2]
```diff
@@ -650,6 +650,23 @@ Fetches information of an Instance of a Component.
...
+
+```
+# simple example
+serverless test --inputs domain=serverless.com
+
+# passing objects with JSON syntax
+serverless invoke --inputs env='{"LANG": "en"}'
+
+# passing arrays with comma seperation
+serverless backup --inputs userIds=foo,bar
```
Comment (timqian) [inline README.md:667]: serverless component doesn't have  these methods: `invoke`, `test`, and `backup`, will it be better to avoid these in the examples?

Comment (eahefnawy) [reply README.md:667]: I don't think it matters. It's unlikely we'll find real-world examples that cover each of these use cases. I just want to demo how to use array, json...etc

**Review by timqian — APPROVED** [review]

**Pull request merged**